### PR TITLE
Technical challenge 1 it 2

### PR DESCRIPTION
**Pull request by CristobalMe**
## Description
Adds new metrics to TC 1 formula and converts "recommend events" in to a function
Solved bug: recommending past events
Made searching distance customized to the user, modified Distance and Date score
Moved all the event filtering to the start

Ignore (minor update, changed date for DateTime): 
backend/prisma/migrations/20240717223719_data_model_v6/migration.sql
backend/prisma/schema.prisma
backend/prisma/seed.js
backend/routes/eventRoutes.js

## Milestones
TC 1

## Resources
None

## Test Plan
Edge cases tested: 
**New user (no data)**

### CODE REVIEW

File: backend/routes/userRoutes.js [hunk 1]
```diff
@@ -172,16 +193,25 @@ router.get('/recommendedEvents/:userId', async (req, res) => {
...
     eventDate = new Date(parseInt(e.date.slice(0, 2)) + '/' + parseInt(e.date.slice(3, 5)) + '/' + parseInt(e.date.slice(6, 10)))
     dateDifference =  ( ((eventDate.getTime() - today.getTime()) / 1000) / 604800 ) 
 
-    if (dateDifference !=0 ){
+    if (dateDifference > 0 ){
       dateScore = 1 / dateDifference
     } else{
-      dateScore = 1
+      // This is to not recommend past events to users
+      dateScore = - 1000
```
Comment (toriqyanM) [inline backend/routes/userRoutes.js:214]: Add date filtering in your db query instead of filtering with scores

File: backend/routes/userRoutes.js [hunk 2]
```diff
@@ -160,6 +155,32 @@ router.get('/recommendedEvents/:userId', async (req, res) => {
     }
   })
   // ------------------------------------------------
+  // Get the id's of the events the user has commented on & how many comments the user has
+  const userComments = await prisma.comment.findMany({
+    where: { 
+      userPosting: user.username
+    }
+  })
+  let totalComments = 0
```
Comment (toriqyanM) [inline backend/routes/userRoutes.js:164]: totalComments can be a constant and instantiated by userComments length

File: backend/routes/userRoutes.js [hunk 3]
```diff
@@ -210,12 +240,30 @@ router.get('/recommendedEvents/:userId', async (req, res) => {
...
-    totalScore = dateScore + categoryScore + distanceScore
+    totalScore = dateScore + categoryScore + distanceScore + commentScore + 4*popularityScore
     // -------------------------------------------------------------------------------------------------------------------------
-
-    idRecommendedEvents.push(e.id)
-    scoreRecommendedEvents.push(totalScore)
+    if (!idEventsAttending.includes(e.id)){
+      idRecommendedEvents.push(e.id)
+      scoreRecommendedEvents.push(totalScore)
+    }
```
Comment (toriqyanM) [inline backend/routes/userRoutes.js:266]: Consider moving all the event filtering to the start of the recommendation logic so that we don't need to calculate scores for the events that we know we don't need to recommend.

File: backend/routes/userRoutes.js [hunk 4]
```diff
@@ -105,19 +105,14 @@ router.patch('/location', async(req, res) => {
   res.json(updatedUser)
 })
 
-router.get('/recommendedEvents/:userId', async (req, res) => {
-  const { userId } = req.params
+const recommendTenEvents = async (user) => {
```
Comment (toriqyanM) [inline backend/routes/userRoutes.js:108]: Break each score into a separate function. This function is still too long.

File: backend/routes/userRoutes.js (hunk 2 above)
Comment (CristobalMe) [reply backend/routes/userRoutes.js:176]: Done

File: backend/routes/userRoutes.js (hunk 3 above)
Comment (CristobalMe) [reply backend/routes/userRoutes.js:274]: Done

File: backend/routes/userRoutes.js (hunk 1 above)
Comment (CristobalMe) [reply backend/routes/userRoutes.js:222]: Done

File: backend/routes/userRoutes.js (hunk 4 above)
Comment (CristobalMe) [reply backend/routes/userRoutes.js:108]: Will do in separate PR

Comment (CristobalMe) [reply backend/routes/userRoutes.js:108]: Done
